packages/wdio-browserstack-service/src/testOps/listener.ts

```diff
@@ -0,0 +1,227 @@
+import UsageStats from './usageStats.js'
+import type FeatureStats from './featureStats.js'
+import RequestQueueHandler from '../request-handler.js'
+import type { CBTData, LogData, ScreenshotLog, TestData, UploadType } from '../types.js'
+import { batchAndPostEvents, sleep } from '../util.js'
+import {
+    DATA_BATCH_ENDPOINT,
+    DEFAULT_WAIT_INTERVAL_FOR_PENDING_UPLOADS,
+    DEFAULT_WAIT_TIMEOUT_FOR_PENDING_UPLOADS,
+    LOG_KIND_USAGE_MAP
+} from '../constants.js'
+import { sendScreenshots } from './requestUtils.js'
+import { BStackLogger } from '../bstackLogger.js'
+
+class Listener {
+    private static instance: Listener
+    private readonly usageStats: UsageStats = UsageStats.getInstance()
+    private readonly testStartedStats: FeatureStats = this.usageStats.testStartedStats
+    private readonly testFinishedStats: FeatureStats = this.usageStats.testFinishedStats
+    private readonly hookStartedStats: FeatureStats = this.usageStats.hookStartedStats
+    private readonly hookFinishedStats: FeatureStats = this.usageStats.hookFinishedStats
+    private readonly cbtSessionStats: FeatureStats = this.usageStats.cbtSessionStats
+    private readonly logEvents: FeatureStats = this.usageStats.logStats
+    private requestBatcher?: RequestQueueHandler
+    private pendingUploads = 0
+
+    private constructor() {
+    }
+
```

Comment:
```suggestion
```

Reply:
I wanted to make this class a singleton class, that's why made the constructor private, any issues in that?

Reply:
Let's add a comment then, to document why you doing this.

Reply:
Have added comment in the code